scripts/analyze_codebase.py

```
@@ -0,0 +1,284 @@
+- Duplicate code blocks
+- Unreferenced modules/imports
+"""
+
+import argparse
+import ast
+import hashlib
+import json
+import os
+import sys
+from pathlib import Path
+from typing import Any, Dict, List, Set, Tuple
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)